File: android_apt/app/src/main/java/top/zcwfeng/android_apt/SecondActivity.java
package top.zcwfeng.android_apt;

import android.os.Bundle;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;

import java.util.Arrays;
import java.util.List;

import top.zcwfeng.android_apt.annotation.Autowired;
import top.zcwfeng.android_apt.annotation.IndectUtils;
import top.zcwfeng.android_apt.bean.UserParceable;
import top.zcwfeng.android_apt.bean.UserSerializable;

public class SecondActivity extends AppCompatActivity {
    @Autowired
    String name;
    @Autowired
    public String attr;
    @Autowired
    int[] arry;
    @Autowired
    UserParceable userParceable;
    @Autowired
    UserParceable[] userParceables;
    @Autowired
    List<UserParceable> userParcelableList;
    @Autowired("users")
    UserSerializable[] userSerializables;

    @Override
    public String toString() {
        return "SecondActivity{" +
                "name='" + name + '\'' +
                ", attr='" + attr + '\'' +
                ", arry=" + Arrays.toString(arry) +
                ", userParceable=" + userParceable +
                ", userParceables=" + Arrays.toString(userParceables) +
                ", userParcelableList=" + userParcelableList +
                ", userSerializables=" + Arrays.toString(userSerializables) +
                '}';
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        IndectUtils.injectExtras(this);
        Log.e("zcw::",toString());
    }
}

File: android_hilt/hiltsample/src/main/java/top/zcwfeng/hiltsample/SecondActivity.kt
package top.zcwfeng.hiltsample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import dagger.hilt.android.AndroidEntryPoint
import top.zcwfeng.hiltsample.data.HttpObject
import javax.inject.Inject

@AndroidEntryPoint
class SecondActivity : AppCompatActivity() {
    @Inject
    lateinit var httpObject:HttpObject

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        Log.e("zcwfeng", "${httpObject.toString()}---sec")
    }
}